src/compiler/generate-query-helper.xsl

```diff
@@ -60,12 +60,9 @@
          </xsl:when>
          <xsl:when test="node()">
             <xsl:text> := ( document {</xsl:text>
-            <xsl:for-each select="node() except text()[not(normalize-space(.))]">

```

Comment:
It's intentional that the replacement code does not make an exception for whitespace-only text nodes, right? I suspect that your code is correct, but I wanted to ask just to make sure it wasn't an inadvertent omission.

Reply:
Nice catch. I can't recall the rationale of the omission 🙄, which means this PR lacks something. I'll look into it.

Reply:
The omission is correct. Whitespace-only text nodes should not be handled specially here.
I guess the exception of whitespace-only text node should have been regarded as a bug when it was first introduced 9 years ago (044446c8b2b8118a418da1cfd4494cd987941a97), because
* [The XSLT counterpart](https://github.com/xspec/xspec/blob/044446c8b2b8118a418da1cfd4494cd987941a97/generate-tests-helper.xsl#L59) did not do so.
* Whitespace-only text nodes were handled specially [before](https://github.com/xspec/xspec/blob/044446c8b2b8118a418da1cfd4494cd987941a97/generate-common-tests.xsl#L89-L98) and [after](https://github.com/xspec/xspec/blob/044446c8b2b8118a418da1cfd4494cd987941a97/generate-xspec-tests.xsl#L420-L422) entering the code in question. 


I submitted #571 which adds a comment to clarify how whiltespace-only text node is handled.
I also added a whitespace-only test node test which *potentially* hit the code in question. (Actually the added test doesn't hit `except text()[not(normalize-space(.))]`, because the current XSpec code [handles](https://github.com/xspec/xspec/blob/5565ad4913c5f761d49b214f10108dc3bc67275b/src/compiler/generate-common-tests.xsl#L122-L144) whitespace-only text nodes before applying `mode="test:generate-variable-declarations"`.)

Reply:
Sounds great. Thanks for checking this and explaining it.